src/crimson/os/alienstore/thread_pool.h

```diff
@@ -72,21 +72,29 @@ struct SubmitQueue {
   }
 };
 
+struct ShardedWorkQueue {
+  std::mutex mutex;
+  std::condition_variable cond;
+  std::deque<WorkItem*> pending;
+  void reserve(size_t queue_size) {
+    pending.resize(queue_size);
```

Comment:
i don't understand why we need to prefill `pending` with nullptrs.

Reply:
@xxhdx1985126 could you please help me understand this part? as `deque::resize()` not only change the container's capacity, the method also appends the default-initialized elements to the deque.

Reply:
oh, sorry, I wanted to follow the previous "pending(queue_size)" initialization, but I failed to notice that this would lead to creation of elements. Will remove it:-)